example-get-started-experiments/code/.github/workflows/deploy-model.yml

```diff
@@ -0,0 +1,57 @@
+name: Deploy model
+
+on:
+  push:
+    # When a new version is registered in Studio Model Registry
+    tags:
+      - "results/train=pool-segmentation@*"
+
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Manual version name'
+        required: true
+        type: string
+  
+permissions:
+  contents: write
+  id-token: write
+
+jobs:
+  deploy-model:
+    environment: cloud
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+    - id: clean_version_name
+      run: echo "version=$(echo ${{ github.ref_name || github.event.inputs.version }} | sed 's/[^a-zA-Z0-9-]/-/g')" >> "$GITHUB_OUTPUT"
+
+    - uses: aws-actions/configure-aws-credentials@v1
+      with:
+        aws-region: us-east-2
+        role-to-assume: arn:aws:iam::342840881361:role/SandboxUser
+        role-duration-seconds: 43200
+
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.8'
+        cache: 'pip'
+        cache-dependency-path: requirements.txt
+
+    - run: pip install -r requirements.txt
+
+    - run:  dvc remote add -d --local storage s3://dvc-public/remote/get-started-pools
+
+    - env:
```

Comment:
@daavoo I've added suggestion about using the GTO action to find the event, version and stage and use it to create the deployment, so that there can be different models/configs per version/stage and different endpoints per stage.

Reply:
I updated the workflow to parse the GTO in a separate job, so we don't do any unnecessary steps in case it is not an assignment event